htdocs/core/boxes/box_birthdays.php

```diff
@@ -85,19 +85,19 @@ public function loadBox($max = 20)
 		if ($user->rights->user->user->lire) {
 			$tmparray = dol_getdate(dol_now(), true);
 
-			$sql = "SELECT u.rowid, u.firstname, u.lastname, u.birth as datea, 'birth' as typea, u.email, u.statut as status";
+			$sql = "SELECT u.rowid, u.firstname, u.lastname, u.birth as datea, date_format(u.birth, '%d') as daya, 'birth' as typea, u.email, u.statut as status";
 			$sql .= " FROM ".MAIN_DB_PREFIX."user as u";
 			$sql .= " WHERE u.entity IN (".getEntity('user').")";
 			$sql .= " AND u.statut = 1";
 			$sql .= dolSqlDateFilter('u.birth', 0, $tmparray['mon'], 0);
 			$sql .= ' UNION ';
-			$sql .= "SELECT u.rowid, u.firstname, u.lastname, u.dateemployment as datea, 'employment' as typea, u.email, u.statut as status";
+			$sql .= "SELECT u.rowid, u.firstname, u.lastname, u.dateemployment as datea, date_format(u.dateemployment, '%d') as daya, 'employment' as typea, u.email, u.statut as status";
 			$sql .= " FROM ".MAIN_DB_PREFIX."user as u";
 			$sql .= " WHERE u.entity IN (".getEntity('user').")";
 			$sql .= " AND u.statut = 1";
 			$sql .= dolSqlDateFilter('u.dateemployment', 0, $tmparray['mon'], 0);
 
-			$sql .= " ORDER BY DAY(datea) ASC";
+			$sql .= " ORDER BY daya ASC";
```

Comment:
Why not just doing ORDER BY datea ?
Adding a field "date_format(u.birth, '%d') as daya" to make an order means the database will not be able to use the index.

Reply:
That was my initial intuition as well (see https://github.com/Dolibarr/dolibarr/commit/8be998a9066d5f5c9d3282f106f8c9f81e833357) but after asking the original author (https://github.com/Dolibarr/dolibarr/issues/24138#issuecomment-1472074229) he mentioned that the birthday box needs to specifically sort on the birth day, while sorting on `datea` would sort people born in earlier years first, even if their actual day of birth is later.

Reply:
As far as I can tell, and I may mistaken, DAY(datea) on MySQL would also need to apply a SQL function over the whole index, then sort that result. So if we both want to fix SQL strictness *and* improve performance by sorting on an indexed field, then we probably should add an index to the schema that contains the day number of both the birthday and the employment date.

Changing the schema is however beyond my current capabilities and familiarity with the code, but I can give it a try if you deem it necessary !